OpenRA.Mods.Common/Traits/Buildings/Exit.cs

```diff
@@ -46,16 +46,14 @@ public Exit(ActorInitializer init, ExitInfo info)
 
 	public static class ExitExts
 	{
-		public static Exit FirstExitOrDefault(this Actor actor, string productionType = null)
+		public static Exit NearestExitOrDefault(this Actor actor, string productionType = null, Func<Exit, bool> p = null, WPos? nearestTo = null)
```

Comment:
ah, so my main point above was to allow us to change this to
```csharp
public static Exit NearestExitOrDefault(this Actor actor, WPos pos, string productionType = null, Func<Exit, bool> p = null)
{
	var all = Exits(actor, productionType)
		.OrderBy(e => e.Info.Priority)
		.ThenBy(e => (actor.CenterPosition + e.Info.SpawnOffset - pos).Length);

	return p != null ? all.FirstOrDefault(p) : all.FirstOrDefault();
}
```
😅

Reply:
I see, fixed hopefully.